tests/testthat/test-permutation_inference.R

```diff
@@ -0,0 +1,91 @@
+
+
+library( tidyverse )
+
+data(basque,  package="Synth")
+
+head( basque)
+
+#basque <- basque %>%
+#    dplyr::select( regionno, year, gdpcap ) %>%
+#    as_tibble()
+
+basque <- basque %>%
+    mutate(trt = case_when(year < 1975 ~ 0,
+                           regionno != 17 ~ 0,
+                           regionno == 17 ~ 1)) %>%
+    filter(regionno != 1) %>%
+    dplyr::select( -c( sec.agriculture:invest ) )
+
+head( basque )
+
+table( basque$trt, basque$regionno )
+
+
+
+test_that( "MDES_table corresponds to default treatment table", {
+
+    syn <- augsynth(gdpcap ~ trt, regionno, year,
+                    data=basque, scm = TRUE)
+
+    summ <- summary(syn, inf_type = 'permutation')
+
+    mm <- summ$permutations$MDES_table[1:7] %>%
+        select( sort( names(.)))
+    mm
+    tt <- treated_table(syn) %>%
+        select( sort( names(.)))
+    tt
+
+    expect_equal( as.data.frame(tt)[c("ATT","raw_average","Yhat", "tx")],
+                  as.data.frame(mm)[c("ATT","raw_average","Yhat", "tx")] )
+} )
+
+test_that( "Placebo distribtion works", {
+
+    syn <- augsynth(gdpcap ~ trt, regionno, year,
+                    data=basque, scm = TRUE, progfunc= "none" )
+    tt <- treated_table(syn)
+    tt
+
+    donor_table(syn)
+
+    b2 = basque %>%
+        mutate( trt = 0 + (regionno == 7 ) * (year>=1975) )
+    syn7 <- augsynth(gdpcap ~ trt, regionno, year,
+                    data=b2, scm = TRUE, progfunc= "none")
+
+    # These should be the same?  Or close to the same?
+    # (They were not under ridge, probably due to the cross-validation procedure.)
+    gaps7 <- augsynth:::get_placebo_gaps(syn7)
```

Comment:
Can you test this with ridge with a fixed lambda parameter? If they should be same then the test should pass.